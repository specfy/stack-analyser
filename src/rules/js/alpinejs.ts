import { register } from '../../register.js';

register({
  tech: 'alpinejs',
  name: 'Alpine.js',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'alpinejs' }],
});
