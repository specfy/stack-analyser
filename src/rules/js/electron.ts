import { register } from '../../register.js';

register({
  tech: 'electron',
  name: 'Electron',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'electron' }],
});
