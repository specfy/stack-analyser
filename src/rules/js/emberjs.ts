import { register } from '../../register.js';

register({
  tech: 'emberjs',
  name: 'Ember',
  type: 'tool',
  dependencies: [
    { type: 'npm', name: 'ember-cli' },
    { type: 'npm', name: 'ember' },
    { type: 'npm', name: 'ember-data' },
    { type: 'npm', name: 'ember-source' },
  ],
});
