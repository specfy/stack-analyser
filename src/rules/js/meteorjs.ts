import { register } from '../../register.js';

register({
  tech: 'meteorjs',
  name: 'Meteor',
  type: 'tool',
  files: ['.meteor'],
  dependencies: [
    { type: 'npm', name: 'meteor' },
    { type: 'npm', name: 'meteor-node-stubs' },
  ],
});
