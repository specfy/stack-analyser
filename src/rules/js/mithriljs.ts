import { register } from '../../register.js';

register({
  tech: 'mithriljs',
  name: 'Mithril',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'mithril' }],
});
