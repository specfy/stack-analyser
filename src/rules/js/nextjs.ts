import { register } from '../../register.js';

register({
  tech: 'nextjs',
  name: 'Next.js',
  type: 'tool',
  files: ['next.config.js'],
  dependencies: [{ type: 'npm', name: 'next' }],
});
