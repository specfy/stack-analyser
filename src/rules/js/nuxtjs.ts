import { register } from '../../register.js';

register({
  tech: 'nuxtjs',
  name: 'Nuxt.js',
  type: 'tool',
  files: ['nuxt.config.js'],
  dependencies: [{ type: 'npm', name: 'nuxt' }],
});
