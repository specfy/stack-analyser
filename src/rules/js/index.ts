import './alpinejs.js';
import './angular.js';
import './bootstrap.js';
import './emberjs.js';
import './esbuild.js';
import './eslint.js';
import './express.js';
import './fastify.js';
import './koa.js';
import './meteorjs.js';
import './mithriljs.js';
import './nextjs.js';
import './nuxtjs.js';
import './preactjs.js';
import './prettier.js';
import './prisma.js';
import './puppeteer.js';
import './react.js';
import './reactEmail.js';
import './rollup.js';
import './sequelize.js';
import './socketio.js';
import './solidjs.js';
import './storybook.js';
import './strapi.js';
import './stylelint.js';
import './sveltejs.js';
import './tailwind.js';
import './typescript.js';
import './vite.js';
import './vue.js';
import './webpack.js';
