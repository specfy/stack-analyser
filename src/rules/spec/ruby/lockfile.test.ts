import { describe, expect, it } from 'vitest';

import { analyser } from '../../../analyser/index.js';
import { rawList } from '../../../loader.js';
import { flatten } from '../../../payload/helpers.js';
import { FakeProvider } from '../../../provider/fake.js';

import '../../../autoload.js';

describe('ruby (component)', () => {
  it('should match everything', async () => {
    const lockfile: string[] = [
      `# frozen_string_literal: true

source "https://rubygems.org"
`,
    ];

    for (const item of rawList) {
      if (item.type !== 'dependency' || item.ref.type !== 'ruby') {
        continue;
      }

      const example = 'example' in item.ref ? item.ref.example : item.ref.name;
      lockfile.push(`gem "${example}", "1.0.0"\n`);
    }

    const res = await analyser({
      provider: new FakeProvider({
        paths: {
          '/': ['Gemfile'],
        },
        files: {
          '/Gemfile': lockfile.join(''),
        },
      }),
    });

    const merged = flatten(res, { merge: true });
    expect(Array.from(merged.techs).sort()).toMatchSnapshot();
    expect(Array.from(merged.dependencies).sort()).toMatchSnapshot();
  });
});
