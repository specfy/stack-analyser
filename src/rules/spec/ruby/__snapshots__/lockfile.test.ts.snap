// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ruby (component) > should match everything 1`] = `
[
  "algolia",
  "auth0",
  "aws",
  "aws.dynamodb",
  "aws.ec2",
  "aws.ecs",
  "aws.elasticache",
  "aws.glacier",
  "aws.lambda",
  "aws.rds",
  "aws.s3",
  "aws.sns",
  "aws.sqs",
  "azure",
  "brevo",
  "cloudflare",
  "couchbase",
  "datadog",
  "docusign",
  "elasticsearch",
  "fastly",
  "firebase",
  "gcp",
  "gcp.aiplatform",
  "gcp.bigquery",
  "gcp.bigtable",
  "gcp.cloudrun",
  "gcp.dataflow",
  "gcp.dataproc",
  "gcp.datastore",
  "gcp.dialogflow",
  "gcp.dns",
  "gcp.functions",
  "gcp.gce",
  "gcp.gcs",
  "gcp.kms",
  "gcp.language",
  "gcp.maps",
  "gcp.memorystore",
  "gcp.pubsub",
  "gcp.secretmanager",
  "gcp.spanner",
  "gcp.speech",
  "gcp.sql",
  "gcp.tasks",
  "gcp.translate",
  "gcp.vision",
  "github",
  "hashicorp_vault",
  "heroku",
  "influxdb",
  "kubernetes",
  "launchdarkly",
  "mailchimp",
  "mailjet",
  "mariadb",
  "meilisearch",
  "memcached",
  "mongodb",
  "mysql",
  "neo4j",
  "newrelic",
  "okta",
  "oraclecloud",
  "pagerduty",
  "postgresql",
  "rabbitmq",
  "redis",
  "ruby",
  "sentry",
  "slack",
  "snowflake",
  "splitio",
  "sqlite",
  "sqreen",
  "zookeeper",
]
`;

exports[`ruby (component) > should match everything 2`] = `
[
  [
    "ruby",
    "MailchimpMarketing",
    "1.0.0",
  ],
  [
    "ruby",
    "MailchimpTransactional",
    "1.0.0",
  ],
  [
    "ruby",
    "algolia",
    "1.0.0",
  ],
  [
    "ruby",
    "algoliasearch-rails",
    "1.0.0",
  ],
  [
    "ruby",
    "auth0",
    "1.0.0",
  ],
  [
    "ruby",
    "aws-sdk-dynamodbstreams",
    "1.0.0",
  ],
  [
    "ruby",
    "aws-sdk-e2",
    "1.0.0",
  ],
  [
    "ruby",
    "aws-sdk-ec2",
    "1.0.0",
  ],
  [
    "ruby",
    "aws-sdk-ecs",
    "1.0.0",
  ],
  [
    "ruby",
    "aws-sdk-elasticache",
    "1.0.0",
  ],
  [
    "ruby",
    "aws-sdk-elasticsearchservice",
    "1.0.0",
  ],
  [
    "ruby",
    "aws-sdk-glacier",
    "1.0.0",
  ],
  [
    "ruby",
    "aws-sdk-lambda",
    "1.0.0",
  ],
  [
    "ruby",
    "aws-sdk-rds",
    "1.0.0",
  ],
  [
    "ruby",
    "aws-sdk-s3",
    "1.0.0",
  ],
  [
    "ruby",
    "aws-sdk-sns",
    "1.0.0",
  ],
  [
    "ruby",
    "aws-sdk-sqs",
    "1.0.0",
  ],
  [
    "ruby",
    "azure-armrest",
    "1.0.0",
  ],
  [
    "ruby",
    "browserstack-local",
    "1.0.0",
  ],
  [
    "ruby",
    "bunny",
    "1.0.0",
  ],
  [
    "ruby",
    "circleci",
    "1.0.0",
  ],
  [
    "ruby",
    "cloudflare-rails",
    "1.0.0",
  ],
  [
    "ruby",
    "couchbase",
    "1.0.0",
  ],
  [
    "ruby",
    "cypress-on-rails",
    "1.0.0",
  ],
  [
    "ruby",
    "dalli",
    "1.0.0",
  ],
  [
    "ruby",
    "datadog-lambda",
    "1.0.0",
  ],
  [
    "ruby",
    "ddtrace",
    "1.0.0",
  ],
  [
    "ruby",
    "dependabot-common",
    "1.0.0",
  ],
  [
    "ruby",
    "docusign_esign",
    "1.0.0",
  ],
  [
    "ruby",
    "elasticsearch",
    "1.0.0",
  ],
  [
    "ruby",
    "elastimeilisearchcsearch",
    "1.0.0",
  ],
  [
    "ruby",
    "fastlane-plugin-firebase_app_distribution",
    "1.0.0",
  ],
  [
    "ruby",
    "fastly",
    "1.0.0",
  ],
  [
    "ruby",
    "functions_framework ",
    "1.0.0",
  ],
  [
    "ruby",
    "gitlab",
    "1.0.0",
  ],
  [
    "ruby",
    "google-apis-sqladmin_v1beta4",
    "1.0.0",
  ],
  [
    "ruby",
    "google-cloud-ai_platform-v1",
    "1.0.0",
  ],
  [
    "ruby",
    "google-cloud-bigquery",
    "1.0.0",
  ],
  [
    "ruby",
    "google-cloud-bigtable",
    "1.0.0",
  ],
  [
    "ruby",
    "google-cloud-build-v1",
    "1.0.0",
  ],
  [
    "ruby",
    "google-cloud-compute",
    "1.0.0",
  ],
  [
    "ruby",
    "google-cloud-dataflow-v1",
    "1.0.0",
  ],
  [
    "ruby",
    "google-cloud-dataproc-v1",
    "1.0.0",
  ],
  [
    "ruby",
    "google-cloud-datastore",
    "1.0.0",
  ],
  [
    "ruby",
    "google-cloud-dialogflow-v1",
    "1.0.0",
  ],
  [
    "ruby",
    "google-cloud-dns",
    "1.0.0",
  ],
  [
    "ruby",
    "google-cloud-functions-v1",
    "1.0.0",
  ],
  [
    "ruby",
    "google-cloud-kms-v1",
    "1.0.0",
  ],
  [
    "ruby",
    "google-cloud-language-v1",
    "1.0.0",
  ],
  [
    "ruby",
    "google-cloud-pubsub",
    "1.0.0",
  ],
  [
    "ruby",
    "google-cloud-redis-v1",
    "1.0.0",
  ],
  [
    "ruby",
    "google-cloud-run-v1",
    "1.0.0",
  ],
  [
    "ruby",
    "google-cloud-run_v2",
    "1.0.0",
  ],
  [
    "ruby",
    "google-cloud-secret_manager-v1",
    "1.0.0",
  ],
  [
    "ruby",
    "google-cloud-spanner-v1",
    "1.0.0",
  ],
  [
    "ruby",
    "google-cloud-speech-v1",
    "1.0.0",
  ],
  [
    "ruby",
    "google-cloud-storage",
    "1.0.0",
  ],
  [
    "ruby",
    "google-cloud-tasks-v1",
    "1.0.0",
  ],
  [
    "ruby",
    "google-cloud-translate-v1",
    "1.0.0",
  ],
  [
    "ruby",
    "google-cloud-vision-v1",
    "1.0.0",
  ],
  [
    "ruby",
    "google_maps_service",
    "1.0.0",
  ],
  [
    "ruby",
    "helm-rb",
    "1.0.0",
  ],
  [
    "ruby",
    "influxdb-client",
    "1.0.0",
  ],
  [
    "ruby",
    "jenkins_api_client",
    "1.0.0",
  ],
  [
    "ruby",
    "kubeclient",
    "1.0.0",
  ],
  [
    "ruby",
    "launchdarkly-server-sdk",
    "1.0.0",
  ],
  [
    "ruby",
    "ldclient-rb",
    "1.0.0",
  ],
  [
    "ruby",
    "mailjet",
    "1.0.0",
  ],
  [
    "ruby",
    "mongo",
    "1.0.0",
  ],
  [
    "ruby",
    "mysql2",
    "1.0.0",
  ],
  [
    "ruby",
    "neo4j-ruby-driver",
    "1.0.0",
  ],
  [
    "ruby",
    "newrelic_rpm",
    "1.0.0",
  ],
  [
    "ruby",
    "oci",
    "1.0.0",
  ],
  [
    "ruby",
    "octokit",
    "1.0.0",
  ],
  [
    "ruby",
    "oktakit",
    "1.0.0",
  ],
  [
    "ruby",
    "omniauth-auth0",
    "1.0.0",
  ],
  [
    "ruby",
    "pagerduty",
    "1.0.0",
  ],
  [
    "ruby",
    "pg",
    "1.0.0",
  ],
  [
    "ruby",
    "platform-api",
    "1.0.0",
  ],
  [
    "ruby",
    "redis",
    "1.0.0",
  ],
  [
    "ruby",
    "redis-client",
    "1.0.0",
  ],
  [
    "ruby",
    "ruby-mysql",
    "1.0.0",
  ],
  [
    "ruby",
    "sentry-rails",
    "1.0.0",
  ],
  [
    "ruby",
    "sequel-snowflake",
    "1.0.0",
  ],
  [
    "ruby",
    "sib-api-v3-sdk",
    "1.0.0",
  ],
  [
    "ruby",
    "slack-ruby-client",
    "1.0.0",
  ],
  [
    "ruby",
    "splitclient-rb",
    "1.0.0",
  ],
  [
    "ruby",
    "sqlite3",
    "1.0.0",
  ],
  [
    "ruby",
    "sqreen",
    "1.0.0",
  ],
  [
    "ruby",
    "tencentcloud-sdk-mariadb",
    "1.0.0",
  ],
  [
    "ruby",
    "travis",
    "1.0.0",
  ],
  [
    "ruby",
    "vault",
    "1.0.0",
  ],
  [
    "ruby",
    "zookeeper",
    "1.0.0",
  ],
]
`;
